### PR TITLE
Add nucleo-h743zi-baremetal boilerplate, with HW test

### PR DESCRIPTION
**Pull request by cpq**

### CODE REVIEW

File: examples/stm32/nucleo-h743zi-baremetal/Makefile [hunk 1]
```diff
@@ -0,0 +1,38 @@
+CFLAGS  ?=  -W -Wall -Wextra -Werror -Wundef -Wshadow -Wdouble-promotion \
+						-Wformat-truncation -fno-common -Wconversion -Wno-sign-conversion \
+						-g3 -Os -ffunction-sections -fdata-sections \
+            -I. -Ih7/Include -Icmsis/CMSIS/Core/Include \
+            -mcpu=cortex-m7 -mthumb -mfloat-abi=soft \
```
Comment (scaprile) [inline examples/stm32/nucleo-h743zi-baremetal/Makefile:5]: TODO(): hard FPU

File: examples/stm32/nucleo-h743zi-baremetal/main.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  mg_log_set(MG_LL_DEBUG);  // Set log level
+  mg_timer_add(&mgr, BLINK_PERIOD_MS, MG_TIMER_REPEAT, timer_cb, NULL);
+
+  MG_INFO(("Starting event loop"));
+  for (;;) {
+    mg_mgr_poll(&mgr, 0);
+  }
+
+  return 0;
+}
```
Comment (scaprile) [inline examples/stm32/nucleo-h743zi-baremetal/main.c:40]: No CMSIS ?

**Review by scaprile — APPROVED** [review]
Sadly I don't have a board to help

File: examples/stm32/nucleo-h743zi-baremetal/Makefile (hunk 1 above)
Comment (cpq) [reply examples/stm32/nucleo-h743zi-baremetal/Makefile:5]: Yeah, there is FPU and other TODOs in the mcu.h

File: examples/stm32/nucleo-h743zi-baremetal/main.c (hunk 2 above)
Comment (cpq) [reply examples/stm32/nucleo-h743zi-baremetal/main.c:40]: Well, we could pull both ARM Core cmsis headers and stm32h74x headers.
But that would make this example different from F4 and F7, which do not use CMSIS headers.
Shall we use CMSIS for all cases?

**Pull request merged**